src/bun.js/webcore/body.zig

```diff
@@ -842,7 +843,16 @@ pub const Body = struct {
                 }
 
                 if (locked.readable.get()) |readable| {
-                    readable.done(global);
+                    if (readable.ptr == .Bytes) {
+                        // we are streaming so we inform the error to it
+                        readable.ptr.Bytes.onData(
+                            .{
+                                .err = .{ .JSValue = error_instance },
+                            },
+                            bun.default_allocator,
+                        );
+                    }
+                    readable.abort(global);
```

Comment:
I think just calling `abort` should be enough

Reply:
abort/cancel actually was only canceling the stream not the source, so I fixed that now and removed the onData to emit error.
onCancel actually only mark as done, not actually emits a error so I changed the test that I add to match this expectations.